python/publish/retry.py

```diff
@@ -63,6 +63,32 @@ def increment(self,
                         if message.startswith('api rate limit exceeded') or \
                                 message.endswith('please wait a few minutes before you try again.'):
                             logger.info(f'Response body indicates retry-able error: {message}')
+                            for header in ['X-RateLimit-Limit', 'X-RateLimit-Remaining', 'X-RateLimit-Reset',
+                                           'X-RateLimit-Used', 'X-RateLimit-Resource']:
+                                if header in response.headers:
+                                    value = response.headers.get(header)
+                                    logger.debug(f'Response header contains {header}={value}')
+
+                            # backoff until X-RateLimit-Reset
+                            if 'X-RateLimit-Reset' in response.headers:
+                                value = response.headers.get('X-RateLimit-Reset')
```

Comment:
What if this is in one day?

Reply:
Then it would wait for a day and the workflow would timeout before. But the action would not be able to complete anyway.

The search rate limit has a one minute window, so it would wait at most a minute here. There are hourly limits so the action could be stuck for at most an hour. In same workflow that is still better than rerunning a whole workflow because this action failed on a rate limit. In that case I would say: Don't overuse your limit, then this action works snappy.